src/main/java/org/folio/linked/data/repo/ResourceRepository.java

```diff
@@ -31,4 +32,6 @@ public interface ResourceRepository extends JpaRepository<Resource, Long> {
   @Modifying
   @Query("update Resource r set r.indexDate = current_timestamp() where r.id in (:ids)")
   void updateIndexDateBatch(@Param("ids") Set<Long> ids);
+
+  Optional<Resource> findByInstanceMetadataInventoryId(String inventoryId);
```

Comment:
Is there a way to just retrieve resourceId from the DB? (I think returning Resource will cause unwanted joins with some tables like resource_type etc)

Reply:
Maybe create an InstanceMetadataRespository and provide a method to fetch resourceId from instance Id.

Reply:
`inventory_id` is not indexed in the `instance_metadata` table. I think we have to create an index to make the retrieval fast. Can we add a `create index if not exists  ...` to `instance_metadata.sql` file?

Reply:
done

also we should keep in mind that db schema should be recreated when deploying these changes since the ```instance_metadata.sql``` file has been changed